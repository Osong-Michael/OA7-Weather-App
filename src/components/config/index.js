import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import weatherReducer from '../Redux/reducers';

export const storeFactory = (initialState) => {
  return createStore(weatherReducer, {...initialState}, applyMiddleware(thunk));
}

export const findByTestId = (wrapper, val) => {
  return wrapper.find(`[data-test-id="${val}"]`);
}