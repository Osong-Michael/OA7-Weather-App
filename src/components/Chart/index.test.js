import { shallow } from 'enzyme';
import VerticalBar from './';
import { findByTestId } from '../config';
import data from '../../test_data/data.json';

const prop = data.list.slice(0, 8);


describe('Bar Chart Renders correctly', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(< VerticalBar  data={prop} />);
  });

  test('renders the chart div', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('shows the title of the chart', () => {
    const description = findByTestId(wrapper, 'chart-title');
    expect(description.text()).toEqual('Hourly Data for Wednesday');
  });

});