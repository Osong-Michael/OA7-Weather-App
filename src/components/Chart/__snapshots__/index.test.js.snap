// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar Chart Renders correctly renders the chart div 1`] = `ShallowWrapper {}`;
